packages/svelte2tsx/svelte-jsx.d.ts

```diff
@@ -250,7 +291,65 @@
       // Transition Events
       ontransitionend?: TransitionEventHandler<T>;
       ontransitionendcapture?: TransitionEventHandler<T>;
-    }
+
+      // HashChange Events
+      onhashchange?: HashChangeEventHandler<T>;
+      onhashchangecapture?: HashChangeEventHandler<T>;
+
+      // PopState Events
+      onpopstate?: PopStateEventHandler<T>;
+      onpopstatecapture?: PopStateEventHandler<T>;
+
+      // Device Motion & Orientation Events
+      ondevicemotion?: DeviceMotionEventHandler<T>;
+      ondevicemotioncapture?: DeviceMotionEventHandler<T>;
+      ondeviceorientation?: DeviceOrientationEventHandler<T>;
+      ondeviceorientationcapture?: DeviceOrientationEventHandler<T>;
+      ondeviceorientationabsolute?: DeviceOrientationEventHandler<T>;
+      ondeviceorientationabsolutecapture?: DeviceOrientationEventHandler<T>;
+
+      // Message Events
+      onmessage?: MessageEventHandler<T>;
+      onmessagecapture?: MessageEventHandler<T>;
+      onmessageerror?: MessageEventHandler<T>;
+      onmessageerrorcapture?: MessageEventHandler<T>;
+
+      // PromiseRejection Events
+      onunhandledrejection?: PromiseRejectionEventHandler<T>;
+      onunhandledrejectioncapture?: PromiseRejectionEventHandler<T>;
+      onrejectionhandled?: PromiseRejectionEventHandler<T>;
+      onrejectionhandledcapture?: PromiseRejectionEventHandler<T>;
+
+      // PageTransition Events
+      onpagehide?: PageTransitionEventHandler<T>;
+      onpagehidecapture?: PageTransitionEventHandler<T>;
+      onpageshow?: PageTransitionEventHandler<T>;
+      onpageshowcapture?: PageTransitionEventHandler<T>;
+
+      // Storage Events
+      onstorage?: StorageEventHandler<T>;
+      onstoragecapture?: StorageEventHandler<T>;
+
+      // Global / Window Events
```

Comment:
This part is also covered in #394. But his generic parameter is correct. Maybe you can remove this part and I'll merge both.